static/app/views/insights/common/components/releasesSort.tsx

```diff
@@ -0,0 +1,61 @@
+import {Button} from 'sentry/components/button';
+import {CompositeSelect} from 'sentry/components/compactSelect/composite';
+import {ReleasesSortOption} from 'sentry/constants/releases';
+import {IconSort} from 'sentry/icons';
+import {t} from 'sentry/locale';
+
+export const SORT_BY_OPTIONS = {
+  [ReleasesSortOption.SESSIONS_24_HOURS]: {label: t('Active Sessions')},
+  [ReleasesSortOption.USERS_24_HOURS]: {label: t('Active Users')},
+  [ReleasesSortOption.ADOPTION]: {label: t('Adoption')},
+  [ReleasesSortOption.BUILD]: {label: t('Build Number')},
+  [ReleasesSortOption.DATE]: {label: t('Date Created')},
+  ['number_events']: {label: t('Number of Events')},
+  [ReleasesSortOption.SEMVER]: {label: t('Semantic Version')},
+  [ReleasesSortOption.SESSIONS]: {label: t('Total Sessions')},
+};
+
+export type ReleasesSortByOption = keyof typeof SORT_BY_OPTIONS;
+
+interface Props {
+  environments: string[];
+  onChange: (sortBy: string) => void;
+  sortBy: ReleasesSortByOption;
+}
+
+export function ReleasesSort({environments, sortBy, onChange}: Props) {
+  return (
+    <CompositeSelect
+      trigger={triggerProps => (
+        <Button
+          {...triggerProps}
+          size="xs"
+          icon={<IconSort />}
+          title={t('Sort Releases')}
+          aria-label={t('Sort Releases')}
+        />
+      )}
+    >
+      <CompositeSelect.Region
+        label={t('Sort By')}
+        value={sortBy}
+        onChange={selection => {
+          onChange(selection.value);
+        }}
+        options={Object.entries(SORT_BY_OPTIONS).map(([name, filter]) => {
+          const disabled =
+            name === ReleasesSortOption.ADOPTION && environments.length !== 1;
+
+          return {
+            label: filter.label,
+            value: name,
+            disabled,
```

Comment:
Unless we rename disabled to something that better expressed the intention, it's best to just inline it.
```suggestion
          return {
            label: filter.label,
            value: name,
            disabled: ReleasesSortOption.ADOPTION && environments.length !== 1,
```

Reply:
in any case, I have updated the code